cmd/kubeadm/app/master/selfhosted.go

```diff
@@ -331,7 +323,13 @@ func getSchedulerDeployment(cfg *kubeadmapi.MasterConfiguration) ext.Deployment
 			},
 		},
 	}
-	return d
+}
+
+func prependSelfHostedFlock(name string, args []string) []string {
+	// TODO: flock has to be inside the container, and a shell must exist with this approach
+	// Since the flock binary is present at /bin/flock in a busybox image and /usr/bin/flock in a debian image, we can't hardcode the path
```

Comment:
we might eventually have to provide the flock binary ourselves via a hostpath mount I guess... Right now the built-in busybox image is just too limited...
```
BusyBox v1.26.1 (2017-01-06 18:13:57 UTC) multi-call binary.

Usage: flock [-sxun] FD|{FILE [-c] PROG ARGS}

[Un]lock file descriptor, or lock FILE, run PROG

	-s	Shared lock
	-x	Exclusive lock (default)
	-u	Unlock FD
	-n	Fail rather than wait
```
it doesn't have `--timeout` for instance which we need